sdk/keyvault/azsecrets/autorest.md

```diff
@@ -2,29 +2,79 @@
 
 These settings apply only when `--go` is specified on the command line.
 
-``` yaml
+```yaml
+clear-output-folder: false
+export-clients: true
 go: true
-version: "^3.0.0"
-input-file:
-- https://github.com/Azure/azure-rest-api-specs/blob/e2ef44b87405b412403ccb005bfb3975411adf60/specification/keyvault/data-plane/Microsoft.KeyVault/stable/7.3/secrets.json
+input-file: https://github.com/Azure/azure-rest-api-specs/blob/e2ef44b87405b412403ccb005bfb3975411adf60/specification/keyvault/data-plane/Microsoft.KeyVault/stable/7.3/secrets.json
 license-header: MICROSOFT_MIT_NO_VERSION
-clear-output-folder: false
-output-folder: internal/generated
+module: github.com/Azure/azure-sdk-for-go/sdk/keyvault/azsecrets
+module-version: 0.8.0
 openapi-type: "data-plane"
+output-folder: ../azsecrets
+override-client-name: Client
 security: "AADToken"
-security-scopes:  "https://vault.azure.net/.default"
-use: "@autorest/go@4.0.0-preview.41"
-module-version: 0.8.0
-export-clients: true
+security-scopes: "https://vault.azure.net/.default"
+use: "@autorest/go@4.0.0-preview.42"
+version: "^3.0.0"
 
-# export request creators and response handlers for use by pollers and pagers, and remove the secretVersion path param check
-# (secretVersion == "" is legal for Key Vault but indescribable in OpenAPI)
 directive:
-  - from: keyvault_client.go
+  # make vault URL a parameter of the client constructor
+  - from: swagger-document
+    where: $["x-ms-parameterized-host"]
+    transform: $.parameters[0]["x-ms-parameter-location"] = "client"
+
+  # rename parameter models to match their methods
+  - rename-model:
+      from: SecretRestoreParameters
+      to: RestoreSecretParameters
+  - rename-model:
+      from: SecretSetParameters
+      to: SetSecretParameters
+  - rename-model:
+      from: SecretUpdateParameters
+      to: UpdateSecretParameters
+
+  # delete the Attributes model defined in common.json (it's used only with allOf)
```

Comment:
That is to say, the spec defines this model only so other models can inherit its fields. No operation uses `Attributes`. I imagine we could make the generator recognize that and not output the model. @jhendrixMSFT would that be a task for autorest.go? I imagine we wouldn't want that kind of filtering in autorest core because languages with inheritance may want to use that to model the allOf relationship.